File: requerimiento-frontend/src/app/requerimiento.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})

export class RequerimientoService {

  private baseUrl = 'http://localhost:8080/requerimiento/';

  constructor(private http:HttpClient) { }

  list(): Observable<any> {
    return this.http.get(`${this.baseUrl}`+'list');
  }

  insert(Requerimiento: object): Observable<Object> {
    return this.http.post(`${this.baseUrl}`+'insert', Requerimiento);
  }

  delete(id: number): Observable<any> {
    return this.http.delete(`${this.baseUrl}delete/${id}`, { responseType: 'text' });
  }

  get(id: number): Observable<any> {
    return this.http.get(`${this.baseUrl}get/${id}`);
  }

  update(id: number, value: any): Observable<Object> {
    return this.http.post(`${this.baseUrl}update/${id}`, value);
  }  
}                                           

File: requerimiento-backend/src/main/java/com/requerimiento/app/services/RequerimientoService.java
package com.requerimiento.app.services;

import java.util.List;

import com.requerimiento.app.entities.Requerimiento;

public interface RequerimientoService {
	public List<Requerimiento> list();
    public void insert(Requerimiento requerimiento);
    public void delete(Integer id);
    public void update(Requerimiento requerimiento);
    public Requerimiento get(Integer id);
}
